### PR TITLE
migrate connector_jira to 11.0

### PR DESCRIPTION
**Pull request by guewen**

### CODE REVIEW

File: connector_jira/components/binder.py [hunk 1]
```diff
@@ -1,26 +1,29 @@
...
 
-    _model_name = [
-        'jira.account.analytic.line',
-        'jira.project.project',
-        'jira.project.task',
-        'jira.res.users',
-    ]
+    Where we create an additional model holding the external id.
+    The advantages to have a second models are:
+    * we can link more than 1 JIRA instance to the same record
```
Comment (simahawk) [inline connector_jira/components/binder.py]: I wonder if you ever need this...

File: connector_jira/components/exporter.py [hunk 2]
```diff
@@ -25,15 +26,15 @@
...
-        :param environment: current environment (backend, model_name, ...)
-        :type environment: :py:class:`connector.connector.ConnectorEnvironment`
-        """
-        super(JiraBaseExporter, self).__init__(environment)
+    _name = 'jira.base.exporter'
+    _inherit = ['base.exporter', 'jira.base']
+    _usage = 'record.exporter'
+
+    def __init__(self, work_context):
+        super(JiraBaseExporter, self).__init__(work_context)
```
Comment (simahawk) [inline connector_jira/components/exporter.py]: can we use `super()`? or do you like to ease backward porting of component refactoring?

File: connector_jira/data/cron.xml [hunk 3]
```diff
@@ -3,41 +3,38 @@
 
   <record forcecreate="True" id="ir_cron_jira_import_project_task" model="ir.cron">
     <field name="name">JIRA - Import Project Tasks</field>
+    <field name="model_id" ref="model_jira_backend"/>
+    <field name="code">model._scheduler_import_project_task()</field>
```
Comment (simahawk) [inline connector_jira/data/cron.xml]: must specify `state=code` otherwise it will never run

File: connector_jira/data/cron.xml [hunk 4]
```diff
@@ -3,41 +3,38 @@
...
     <field eval="False" name="doall"/>
-    <field eval="'jira.backend'" name="model"/>
-    <field eval="'_scheduler_import_project_task'" name="function"/>
-    <field eval="'()'" name="args"/>
   </record>
 
   <record forcecreate="True" id="ir_cron_jira_import_res_users" model="ir.cron">
     <field name="name">JIRA - Import Users</field>
+    <field name="model_id" ref="model_jira_backend"/>
+    <field name="code">model._scheduler_import_res_users()</field>
```
Comment (simahawk) [inline connector_jira/data/cron.xml]: same for all of them :)

File: connector_jira/models/project_project/exporter.py [hunk 5]
```diff
@@ -1,35 +1,51 @@
...
-@on_record_create(model_names='jira.project.project')
-@on_record_write(model_names='jira.project.project')
-def delay_export(env, model_name, record_id, vals):
-    common.delay_export(env, model_name, record_id, vals, priority=10)
+class JiraProjectProjectListener(Component):
+    _name = 'jira.project.project.listener'
+    _inherit = ['base.connector.listener']
+    _apply_on = ['jira.project.project']
 
+    @skip_if(lambda self, record, **kwargs: self.no_connector_export(record))
```
Comment (simahawk) [inline connector_jira/models/project_project/exporter.py]: can't this check be moved to base class? I mean, you never want to sync when no export is in place... no?

File: connector_jira/views/project_task_views.xml [hunk 6]
```diff
@@ -60,7 +60,7 @@
         <field name="jira_compound_key"/>
       </field>
 
-      <xpath expr="//div[@class='oe_kanban_content']/div[@class='text-muted']" position="inside">
+      <xpath expr="//small[@class='o_kanban_record_subtitle text-muted']" position="inside">
```
Comment (simahawk) [inline connector_jira/views/project_task_views.xml]: use only `hasclass('o_kanban_record_subtitle')`

File: connector_jira/components/binder.py (hunk 1 above)
Comment (guewen) [reply connector_jira/components/binder.py]: Why not, this is allowed by the design, doesn't mean you have to use it. The point about locks and write on a secondary table alone is a good enough reason.

File: connector_jira/data/cron.xml (hunk 3 above)
Comment (guewen) [reply connector_jira/data/cron.xml]: Erf... thanks, one of the worst change in v11...

File: connector_jira/models/project_project/exporter.py (hunk 5 above)
Comment (guewen) [reply connector_jira/models/project_project/exporter.py]: What base class? Here the decorator is applied on `on_record_create` which can be used for other purposes than syncing/exporting.

Comment (simahawk) [reply connector_jira/models/project_project/exporter.py]: poorly explained myself.... #lazymorning :smile: 
Wouldn't be nice to have `@skip_if_no_export`?
All in all I think we could have a listener mixin that works for all models for exporting. But this is another story.

**Comment by guewen** [comment]
@simahawk updated with your asked changes

**Comment by guewen** [comment]
@simahawk thanks, squashed

I kept some commits apart: "Apply 2to3" because I like to keep a trace of automated changes. "Migrate to new Component API" because it's a good documentation of what needs to be done in such case, the setup guide and the last commit because they may concern the previous version and then are easy to backport.

**Pull request merged**